packages/mui-base/src/Tooltip/Root/TooltipRoot.tsx

```diff
@@ -0,0 +1,156 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import type { TooltipRootProps } from './TooltipRoot.types';
+import { TooltipRootContext } from './TooltipRootContext';
+import { useTooltipRoot } from './useTooltipRoot';
+
+/**
+ * The foundation for building custom-styled tooltips.
+ *
+ * Demos:
+ *
+ * - [Tooltip](https://mui.com/base-ui/react-tooltip/)
+ *
+ * API:
+ *
+ * - [TooltipRoot API](https://mui.com/base-ui/react-tooltip/components-api/#tooltip-root)
+ */
+function TooltipRoot(props: TooltipRootProps) {
+  const {
+    delayType = 'rest',
+    delay = 200,
+    closeDelay = 0,
+    hoverable = true,
+    animated = true,
+    followCursorAxis = 'none',
+  } = props;
+
+  const [triggerElement, setTriggerElement] = React.useState<Element | null>(null);
+  const [popupElement, setPopupElement] = React.useState<HTMLElement | null>(null);
+
+  const {
+    open,
+    setOpen,
+    mounted,
+    setMounted,
+    instantType,
+    getTriggerProps,
+    getRootPopupProps,
+    rootContext,
+    transitionStatus,
+  } = useTooltipRoot({
+    popupElement,
+    triggerElement,
+    hoverable,
+    animated,
+    followCursorAxis,
+    delay,
+    delayType,
+    closeDelay,
+    open: props.open,
+    onOpenChange: props.onOpenChange,
+    defaultOpen: props.defaultOpen,
+  });
+
+  const contextValue = React.useMemo(
+    () => ({
+      delay,
```

Comment:
It might be worth showing a console warning in development when delay-related props are set in both the Root and Group.